### PR TITLE
feat(harvester): harvester push to per-agent API endpoint and add maxFiles configuration

### LINKED ISSUE #25: [Task] Push recordings to per-agent API endpoint
When the Agent pushes JFR file data to Cryostat, it should do so at an API endpoint that corresponds to the specific Agent instance. ie some unique identifier relating to the Agent, which both the Agent and Crysotat agree upon, should be included as an endpoint path parameter or otherwise in the request metadata, so that Cryostat can collect the pushed files and link them to the Agent instance origin for later queries.

Depends on https://github.com/cryostatio/cryostat/issues/1299

### PR DESCRIPTION
**Pull request by maxcao13**
Depends on https://github.com/cryostatio/cryostat/pull/1333
Fixes #25 

1. `mvn clean install` this PR.
3. `cd quarkus-test ; ./mvnw -U clean package && podman build -t quay.io/andrewazores/quarkus-test:latest -f src/main/docker/Dockerfile.jvm . ; podman image prune -f`
4. This PR requires https://github.com/cryostatio/cryostat/pull/1333
5. `cd cryostat`, modify `smoketest.sh` as seen below 
6. Go to the PR linked above and checkout the latest github-actions image linked in the comment and run it
7. e.g. `$ CRYOSTAT_IMAGE=ghcr.io/cryostatio/cryostat:pr-1333-4cb90b936aabdba6487f63bf2001ffcdb23df30f sh smoketest.sh`

```diff
diff --git a/smoketest.sh b/smoketest.sh
index 48c0f88c..c2c6cf45 100755
--- a/smoketest.sh
+++ b/smoketest.sh
@@ -132,7 +132,10 @@ runDemoApps() {
         --env CRYOSTAT_AGENT_BASEURI="${protocol}://localhost:${webPort}/" \
         --env CRYOSTAT_AGENT_TRUST_ALL="true" \
         --env CRYOSTAT_AGENT_AUTHORIZATION="Basic $(echo user:pass | base64)" \
-        --rm -d quay.io/andrewazores/quarkus-test:0.0.10
- 
+        --env CRYOSTAT_AGENT_HARVESTER_MAX_FILES=3 \
+        --env CRYOSTAT_AGENT_HARVESTER_PERIOD_MS=30000 \
+        --env CRYOSTAT_AGENT_HARVESTER_EXIT_MAX_AGE_MS=60000 \
+        --env CRYOSTAT_AGENT_HARVESTER_EXIT_MAX_SIZE_B="$(echo 1024*1024*1 | bc)" \
+        --rm -d quay.io/andrewazores/quarkus-test:latest
 
     podman run \
         --name quarkus-test-agent-2 \
```

Notice the agent will keep the latest 3 files when pushing to its attached target.

One unusual thing I noticed, was that my `quarkus-test` application tests kept failing due to some error with quarkus.
```bash
Running org.acme.GreetingResourceTest
2023-01-16 18:26:28,628 INFO  [org.jbo.threads] (main) JBoss Threads version 3.4.2.Final
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.672 s <<< FAILURE! - in org.acme.GreetingResourceTest
[ERROR] org.acme.GreetingResourceTest.testHelloEndpoint  Time elapsed: 0.005 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Unable to start HTTP server
Caused by: java.util.concurrent.ExecutionException: io.quarkus.runtime.QuarkusBindException
Caused by: io.quarkus.runtime.QuarkusBindException
```
But running the normal steps and just skipping the tests with `-DskipTests` worked fine.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,9 @@ and how it advertises itself to a Cryostat server instance. Required properties
...
 - [ ] `cryostat.agent.harvester.template` [`String`]: the name of the `.jfc` event template configuration to use for the harvester's managed Flight Recording. Default `default`, the continous monitoring event template.
-- [ ] `cryostat.agent.harvester.exit.max-age-ms [`long`]`: the JFR `maxage` setting, specified in milliseconds, to apply to recording data uploaded to the Cryostat server when the JVM this Agent instance is attached to exits. This ensures that tail-end data is captured between the last periodic push and the application exit. Exit uploads only occur when the application receives `SIGINT`/`SIGTERM` from the operating system or container platform.
-- [ ] `cryostat.agent.harvester.exit.max-size-b [`long`]`: the JFR `maxsize` setting, specified in bytes, to apply to exit uploads as described above.
+- [ ] `cryostat.agent.harvester.max-files` [`String`]: the maximum number of files that the agent will keep within the target. The agent will periodically prune, in a FIFO manner, the oldest JFR files within the target's storage, while the number of saved recordings is greater than the maximum file limit. Default `10`.
```
Comment (andrewazores) [inline README.md:49]: This description implies that the recording storage is local to the agent somehow. I think it should be rephrased to be more clear that the configuration still results in the harvester pushing the files over the network, but this setting is used to set a policy on the storage backend for how many of these pushes to retain.

File: src/main/resources/META-INF/microprofile-config.properties [hunk 2]
```diff
@@ -10,5 +10,6 @@ cryostat.agent.webserver.port=9977
 cryostat.agent.app.name=cryostat-agent
 cryostat.agent.harvester.period-ms=-1
 cryostat.agent.harvester.template=default
+cryostat.agent.harvester.max-files=10
```
Comment (andrewazores) [inline src/main/resources/META-INF/microprofile-config.properties:13]: I think the default here should be unlimited, or Integer.MAX_VALUE

**Comment by andrewazores** [comment]
> Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Unable to start HTTP server
Caused by: java.util.concurrent.ExecutionException: io.quarkus.runtime.QuarkusBindException
Caused by: io.quarkus.runtime.QuarkusBindException

This looks like something else was already listening on the same HTTP port that the application wanted to connect to. I think the `quarkus-test` application's unit tests spin up an HTTP server and it tries to bind to the same local port as something in the `smoketest.sh` setup.

**Comment by maxcao13** [comment]
> > Caused by: java.lang.RuntimeException: Failed to start quarkus
> > Caused by: java.lang.RuntimeException: Unable to start HTTP server
> > Caused by: java.util.concurrent.ExecutionException: io.quarkus.runtime.QuarkusBindException
> > Caused by: io.quarkus.runtime.QuarkusBindException
> 
> This looks like something else was already listening on the same HTTP port that the application wanted to connect to. I think the `quarkus-test` application's unit tests spin up an HTTP server and it tries to bind to the same local port as something in the `smoketest.sh` setup.

Ahh... I see. Then that makes sense.

**Comment by andrewazores** [comment]
Looks great otherwise.

**Review by andrewazores — APPROVED** [review]

**Pull request merged**